File: services/nihongo-dico-entries-service/src/main/java/fr/frogdevelopment/nihongo/entries/implementation/search/entity/SearchDetails.java
package fr.frogdevelopment.nihongo.entries.implementation.search.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import java.io.Serializable;
import java.util.Set;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
@JsonInclude(Include.NON_NULL)
public class SearchDetails implements Serializable {

    private static final long serialVersionUID = 3L;

    private long entrySeq;
    private String kanji;
    private String kana;
    private String reading;
    private Set<String> pos;
    private Set<String> field;
    private Set<String> misc;
    private String info;
    private Set<String> dial;
    private String gloss;
}

File: web/nihongo-dico-web/src/app/search-details/SearchDetails.ts
import {Sentence} from './Sentence';

export class SearchDetails {

  kanji: string;
  kana: string;
  reading: string;
  pos: string[];
  field: string[];
  misc: string[];
  info: string;
  dial: string[];
  gloss: string[];

  sentences: Sentence[];
}